tests/kernel/install_kotd.pm

```diff
@@ -7,131 +7,40 @@
-        $url = "$url/$kernel";
-    }
-    else {
-        $url = "$url$kernel";
-    }
-    assert_script_run("wget -O '$file' '$url'", timeout => 1800);
-    return $file;
-}
-
-=head2 install_kernel
-
-  install_kernel($file);
-
-Install kernel file $file with rpm command and C<die> if it returns non-zero value.
-
-=cut
-
-sub install_kernel {
-    my $file = shift;
-    assert_script_run("rpm -i --nodeps --oldpackage --nosignature '$file'", timeout => 120);
-}
-
-# Set KOTD kernel as default boot option
-=head2 set_default
-
-  set_default($file);
-
-Set $file as the default boot option.
-
-=cut
-
-sub set_default {
-    my $file = shift;
-    my $cmd  = <<'END';
-#!/bin/bash
-rpm -q grub2 &> /dev/null
-if [[ $? -eq 0 ]]; then
-    grub_conf='/boot/grub2/grub.cfg'
-else
-    grub_conf='/boot/grub/menu.lst'
-fi
-
-id=$(echo "$1" | awk -F. '{print $(NF-2)}')
-vmlinuz=$(find /boot -name "vmlinuz-*$id*")
-old_version=$(uname -r)
-new_version=$(basename "$vmlinuz" | sed -e 's/vmlinuz-//g')
-sed -ie "s/$old_version/$new_version/g" "$grub_conf"
-END
-    my $script = "set_default.sh";
-    open my $fh, ">", 'current_script' or croak("Could not open file. $!");
-    print $fh $cmd;
-    close $fh;
-    assert_script_run("curl -sfv '" . autoinst_url("/current_script") . "' | bash -xs '$file'");
-}
+use kernel;
+use power_action_utils 'power_action';
 
 sub run {
     my $self = shift;
-
     $self->wait_boot;
     $self->select_serial_terminal;
-
-    my $baseurl = "http://download.suse.de/ibs/Devel:";
-    my $version = "SLE" . get_required_var("VERSION");
-    my $subdir  = "standard";
-    if (is_opensuse) {
-        $baseurl = "http://download.opensuse.org/repositories";
-        $version = "HEAD";
-        # currently only x86_64, i686 and i586 are supported in standard directory
-        $subdir = "ARM" if is_aarch64();
-        $subdir = "PPC" if is_ppc64le();
-    }
-    my $default_url = "$baseurl/Kernel:/$version/$subdir";
-    my $url         = get_var("KOTD_REPO", $default_url);
-    $url .= "/" . get_required_var("ARCH") . "/";
-
-    if (grub_version() == 1) {
-        my $file = download_kernel($url, 'kernel-default-base');
-        install_kernel($file);
-    }
-    my $file = download_kernel($url, "kernel-default");
-    install_kernel($file);
-    set_default($file);
-
-    select_console('root-console');
-    type_string "reboot\n";
+    # Get url of kotd/kmp repositories
+    my $kotd_repo = get_required_var('KOTD_REPO');
+    my $kmp_repo  = get_var('KMP_REPO');
+    # Make sure that system is fully updated
+    fully_patch_system;
+    # Insert isofs module to be able to access repositories on CD after
+    # kernel removal
+    assert_script_run 'modprobe isofs';
+    # Remove all installed kernel and related packages
+    remove_kernel_packages;
+    # Enable kotd/kmp repositories
+    zypper_call("ar -G -f -p 90 $kotd_repo KOTD");
+    zypper_call("ar -G -f -p 90 $kmp_repo KMP") if get_var('KMP_REPO');
```

Comment:
```suggestion
    zypper_call("ar -G -f -p 90 $kmp_repo KMP") if $kmp_repo;
```

$kmp_repo is undef if KMP_REPO is not set

Reply:
I guess `zypper_ar()` is not suitable for you, that's why you use `zypper_call("ar ...")`.

Reply:
@pevik zypper_ar doesn't support priority change and other options

Reply:
@czerw  time to fix `zypper_ar` ?

Reply:
@mimi1vx not now :)

Reply:
Yep, that should be fixed in different PR. Please merge this.

Reply:
`zypper_ar()`: moving discussion to https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/7837#issuecomment-510820271